packages/datatrak-web/src/views/AccountSettingsPage/RequestCountryAccessSection/RequestCountryAccessForm.tsx

```diff
@@ -0,0 +1,244 @@
+    mode: 'onChange',
+  });
+  const {
+    formState: { isSubmitting, isValidating, isValid },
+    handleSubmit,
+    register,
+    reset,
+  } = formContext;
+
+  const applicableCountries =
+    countries?.filter((country: TupaiaWebCountryAccessListRequest.CountryAccess) =>
+      project?.names?.includes(country.name),
+    ) ?? [];
+
+  const formIsNotSubmissible =
+    !project || isValidating || !isValid || isSubmitting || accessListIsLoading || requestIsLoading;
+
+  function onSubmit(formData: RequestCountryAccessFormFields) {
+    requestCountryAccess({
+      entityIds: formData.countryIds,
+      message: formData.reasonForAccess,
+      projectCode: projectCode,
+    });
+  }
+
+  return (
+    <StyledForm formContext={formContext} onSubmit={handleSubmit(onSubmit)}>
+      <StyledFieldset>
+        <CountryChecklistWrapper>
+          <StyledFormLabel>Select countries</StyledFormLabel>
+          <CountryChecklist
+            disabled={!project || isSubmitting || requestIsLoading}
+            key={countryChecklistKey}
+          >
+            {applicableCountries.map(({ id, name, hasAccess, accessRequests }) => {
+              const hasRequestedAccess = accessRequests.includes(ensure(projectCode));
+              const getTooltip = () => {
+                if (hasAccess) return 'You already have access';
+                if (hasRequestedAccess) return 'Approval in progress';
+              };
+
+              return (
+                <StyledCheckbox
+                  color="primary"
+                  disabled={hasAccess || hasRequestedAccess}
+                  id="countryIds"
+                  inputRef={register({ validate: (value: Country['id'][]) => value.length > 0 })}
+                  key={id}
+                  label={name}
+                  name="countryIds"
+                  tooltip={getTooltip()}
+                  value={id}
+                />
+              );
+            })}
+          </CountryChecklist>
+        </CountryChecklistWrapper>
+        <StyledBox>
+          <StyledFormInput
+            disabled={!project}
+            fullWidth
+            Input={TextField}
+            inputProps={{
+              enterKeyHint: 'done',
+              // Make <textarea> scroll upon overflow. MUI uses inline styling (element.style) to
+              // dynamically resize it to fit content, so we must do the same to override it.
+              style: sizeClassIsMdOrLarger ? { height: '100%', overflow: 'scroll' } : {},
+            }}
+            label="Reason for access"
+            margin="none"
+            multiline
+            name="reasonForAccess"
+            size="medium"
+          />
+          <Button
+            disabled={formIsNotSubmissible}
+            tooltip={isValid ? undefined : 'Select countries to request access'}
+            type="submit"
+          >
+            {isSubmitting || requestIsLoading ? 'Submitting request' : 'Request access'}
```

Comment:
It might be nice to move some of these ternaries out from being inline, to make the file a little bit more readable, since there are quite a few elements in here to read